### ISSUE TITLE
Error running CNView

### ISSUE THREAD

**Issue by lindsay-liang** [RCollins13/CNView#10]
I have an input coverage matrix of 176 samples where across a small region; each row in the bedfile covers only 1bp (unlike your example, but the region is pretty small so I'm hoping this won't cause issues).  There are no missing bp intervals in the file for the region.

Running the following:
```
Rscript CNView.R chrX 308758 314496 417_S9  chrX-308758-314496_matrix.txt chrX-308758-314496.pdf
```
Where the matrix looks like:
```
chrom	start	end	100_S2	105_S5 ...
chrX	308605	308606	0	0
chrX	308606	308607	0	0
chrX	308607	308608	1	1
chrX	308608	308609	1	1
chrX	308609	308610	1	1
chrX	308610	308611	1	1
...
```

I get the following error

```
Loading required package: optparse
+-------------------+
| CNView Visualizer |
|     (c) 2017      |
+-------------------+
Sample ID file '100_S2' not found, assuming single sample ID provided
Loading required package: RMySQL
Loading required package: plyr
Loading required package: MASS
Loading required package: metap
Loading required package: Rsamtools
Filtering & loading coverage matrix... Complete
Error in .subset(x, j) : invalid subscript type 'list'
Calls: CNView -> [ -> [.data.frame
Execution halted
```

I saw an old issue posted here: https://github.com/RCollins13/CNView/issues/1 which had the same error but since it was fixed it's likely not to be the same problem.

Thanks for your attention!

**Comment by RCollins13** [comment]
Hi Lindsay,

Happy to help.

At first glance, the input coverage matrix looks like a potential source of the problem. 

CNView expects the first column to be named “Chr”, not “chrom”. I am pretty sure this is what’s causing your problem. Refer to the example coverage matrix provided in the CNView documentation if you’re curious.

If that doesn’t work, here are a few more ideas that may also represent problems:

1. I’ve never tried 1bp intervals before, and CNView wasn’t necessarily designed with this scale of resolution in mind. If you’re interested in per-base coverage plots for small CNVs, I’d recommend something simpler like running a `samtools mpileup` over your region of interest and plotting that.

2. In my recollection, R doesn’t like column names that start with numbers. I thought we patched a fix for this into an earlier release, but you could try appending some string-based prefix, like “sample_”, before your sample IDs in the header column of your coverage matrix.

3. How many bins (rows) in total are in the matrix? CNView run at default parameters needs a large window of bins surrounding your region of interest (this is how the coverage normalization is performed).

4. Those coverage values look unusually low. What data type is this, and how are you collecting coverage? I don’t know what to expect if CNView were fed a matrix of 0’s and 1’s (or extremely sparse counts).

Hope this helps,
Ryan

**Comment by lindsay-liang** [comment]
Hi Ryan, 

Thanks for the response! I made the simple changes in the matrix by changing 'chrom' to 'chr' while also removing 'chr' from the rest of the rows in that column; that seems to have fixed the errors that caused the program to halt.  There was also an issue of R appending an X to the beginning of the sampleID variable, but I just added a line to remove that first character.  

Thanks again!
-Lindsay
